source/maze/MouseController.cpp

```diff
@@ -57,6 +58,12 @@ void MouseController::getDistanceAllCell()
     /* Firstly set the distance of the current position to 0 */
     setDis(getCurrentPos(), CELL_DISTANCE_START);
 
+    /* check if the mouse is in a destination */
+	if (isInDestinationCell()) {
+		destinations.erase(std::remove(destinations.begin(),
+				destinations.end(), getCurrentPos()), destinations.end());
+	}
+
```

Comment:
This should be done after moving the mosue, not at start of running the algorithm. Please move this to `moveNexCell()`.